File: woodroid/src/main/java/me/gilo/woodroid/repo/OrderRepository.java
package me.gilo.woodroid.repo;

import me.gilo.woodroid.data.api.OrderAPI;
import me.gilo.woodroid.models.LineItem;
import me.gilo.woodroid.models.Order;
import me.gilo.woodroid.models.OrderNote;
import me.gilo.woodroid.models.filters.OrderFilter;
import me.gilo.woodroid.repo.order.OrderNoteRepository;
import me.gilo.woodroid.repo.order.RefundRepository;
import retrofit2.Call;

import java.util.List;

public class OrderRepository extends WooRepository {

    private final OrderAPI apiService;

    OrderNoteRepository orderNoteRepository;
    RefundRepository refundRepository;

    public OrderRepository(String baseUrl, String consumerKey, String consumerSecret) {
       super(baseUrl, consumerKey, consumerSecret);
       apiService = retrofit.create(OrderAPI.class);

       orderNoteRepository = new OrderNoteRepository(baseUrl, consumerKey, consumerSecret);
       refundRepository = new RefundRepository(baseUrl, consumerKey, consumerSecret);
    }

    public Call<Order> create(Order order) {
        return apiService.create(order);
    }

    public Call<Order> addToCart(int productId) {
        Order order = new Order();

        LineItem lineItem = new LineItem();
        lineItem.setProductId(productId);
        lineItem.setQuantity(1);

        order.addLineItem(lineItem);

        return apiService.create(order);
    }

    public Call<Order> order(int id) {
        return apiService.view(id);
    }

    public Call<List<Order>> orders() {
        return apiService.list();
    }

    public Call<List<Order>> orders(OrderFilter orderFilter) {
        return apiService.filter(orderFilter.getFilters());
    }

    public Call<Order> update(int id, Order order) {
        return apiService.update(id, order);
    }

    public Call<Order> delete(int id) {
        return apiService.delete(id);
    }

    public Call<Order> delete(int id, boolean force) {
        return apiService.delete(id, force);
    }


    public Call<OrderNote> createNote(Order order, OrderNote note) {
        return orderNoteRepository.create(order, note);
    }

    public Call<OrderNote> note(Order order, int id) {
        return orderNoteRepository.note(order, id);
    }

    public Call<List<OrderNote>> notes(Order order) {
        return orderNoteRepository.notes(order);
    }

    public Call<OrderNote> deleteNote(Order order, int id) {
        return orderNoteRepository.delete(order, id);
    }

    public Call<OrderNote> deleteNote(Order order, int id, boolean force) {
        return orderNoteRepository.delete(order, id, force);
    }


}

File: woodroid/src/main/java/me/gilo/woodroid/repo/OrderRepository.kt
package me.gilo.woodroid.repo

import me.gilo.woodroid.data.api.OrderAPI
import me.gilo.woodroid.models.LineItem
import me.gilo.woodroid.models.Order
import me.gilo.woodroid.models.OrderNote
import me.gilo.woodroid.models.filters.OrderFilter
import me.gilo.woodroid.repo.order.OrderNoteRepository
import me.gilo.woodroid.repo.order.RefundRepository
import retrofit2.Call

class OrderRepository(baseUrl: String, consumerKey: String, consumerSecret: String) :
    WooRepository(baseUrl, consumerKey, consumerSecret) {

    private val apiService: OrderAPI

    internal var orderNoteRepository: OrderNoteRepository
    internal var refundRepository: RefundRepository

    init {
        apiService = retrofit.create(OrderAPI::class.java)

        orderNoteRepository = OrderNoteRepository(baseUrl, consumerKey, consumerSecret)
        refundRepository = RefundRepository(baseUrl, consumerKey, consumerSecret)
    }

    fun create(order: Order): Call<Order> {
        return apiService.create(order)
    }

    fun addToCart(productId: Int, cartOrder: Order?): Call<Order> {
        var cartOrder = cartOrder
        val lineItem = LineItem()
        lineItem.productId = productId
        lineItem.quantity = 1

        if (cartOrder != null) {
            cartOrder.addLineItem(lineItem)
            return apiService.update(cartOrder.id, cartOrder)
        } else {
            cartOrder = Order()
            cartOrder.orderNumber = "Cart"
            cartOrder.status = "on-hold"
            cartOrder.addLineItem(lineItem)
            return apiService.create(cartOrder)
        }

    }

    fun cart(): Call<List<Order>> {
        val orderFilter = OrderFilter()
        orderFilter.status = "on-hold"

        return apiService.filter(orderFilter.filters)
    }

    fun order(id: Int): Call<Order> {
        return apiService.view(id)
    }

    fun orders(): Call<List<Order>> {
        return apiService.list()
    }

    fun orders(orderFilter: OrderFilter): Call<List<Order>> {
        return apiService.filter(orderFilter.filters)
    }

    fun update(id: Int, order: Order): Call<Order> {
        return apiService.update(id, order)
    }

    fun delete(id: Int): Call<Order> {
        return apiService.delete(id)
    }

    fun delete(id: Int, force: Boolean): Call<Order> {
        return apiService.delete(id, force)
    }


    fun createNote(order: Order, note: OrderNote): Call<OrderNote> {
        return orderNoteRepository.create(order, note)
    }

    fun note(order: Order, id: Int): Call<OrderNote> {
        return orderNoteRepository.note(order, id)
    }

    fun notes(order: Order): Call<List<OrderNote>> {
        return orderNoteRepository.notes(order)
    }

    fun deleteNote(order: Order, id: Int): Call<OrderNote> {
        return orderNoteRepository.delete(order, id)
    }

    fun deleteNote(order: Order, id: Int, force: Boolean): Call<OrderNote> {
        return orderNoteRepository.delete(order, id, force)
    }


}
